diff --git a/Development/Src/Tendrils.suo b/Development/Src/Tendrils.suo
index 70897dc..a32238d 100644
Binary files a/Development/Src/Tendrils.suo and b/Development/Src/Tendrils.suo differ
diff --git a/Development/Src/Tendrils/Classes/DrCameraModuleTopDown.uc b/Development/Src/Tendrils/Classes/DrCameraModuleTopDown.uc
index 8e27c95..d8241ca 100644
--- a/Development/Src/Tendrils/Classes/DrCameraModuleTopDown.uc
+++ b/Development/Src/Tendrils/Classes/DrCameraModuleTopDown.uc
@@ -1,7 +1,7 @@
 class DrCameraModuleTopDown extends Object implements(DrCameraModule);
 
 const YawRotRate = 3;
-const PanRate = 1.0;
+const PanRate = 0.7;
 
 var float Height;
 var float HeightTarget;
@@ -29,11 +29,11 @@ simulated function UpdateCamera( Pawn p, DrCamera CamActor, float DT, out TViewT
     if ( Yaw != YawTarget ) {
 		DeltaRot = ( YawTarget - Yaw );
         Yaw +=  ( Abs( DeltaRot ) > class'DrUtils'.const.MAXROT  ?  -DeltaRot : DeltaRot ) * DT * YawRotRate;
-		Yaw = Yaw % ( 2 * class'DrUtils'.const.MAXROT );
+		Yaw = Yaw % ( class'DrUtils'.const.MAXROT );
     }
 
     out_VT.POV.Location = out_VT.Target.Location;
-    out_VT.POV.Location.Z += Height;
+    out_VT.POV.Location.Z = Height;
 
     out_VT.POV.Rotation.Pitch = -class'DrUtils'.const.MAXROT / 2;
     out_VT.POV.Rotation.Yaw = Yaw;
diff --git a/Development/Src/Tendrils/Classes/DrGraphStratSimple.uc b/Development/Src/Tendrils/Classes/DrGraphStratSimple.uc
index 76569f9..7dec09e 100644
--- a/Development/Src/Tendrils/Classes/DrGraphStratSimple.uc
+++ b/Development/Src/Tendrils/Classes/DrGraphStratSimple.uc
@@ -5,6 +5,11 @@ function DrLevel GenLevelGraph( array<DrSection> inSections, delegate<LinkSelect
 	local int iter;
 	local DrLevel out_Level;
 
+    /* Sink all sections down low low low low */
+    for ( iter = 0; iter < inSections.Length; ++iter ) {
+        inSections[iter].SetRelativeLocation( vect( 0, 0, -5000 ) );
+    }
+
 	iter = 0;
 	while ( iter++ < 20 ) {
 		if ( GenIter( inSections, out_Level, LinkSelector ) ) {
diff --git a/Development/Src/Tendrils/Classes/DrPlayerController.uc b/Development/Src/Tendrils/Classes/DrPlayerController.uc
index c2b2683..8c85ed8 100644
--- a/Development/Src/Tendrils/Classes/DrPlayerController.uc
+++ b/Development/Src/Tendrils/Classes/DrPlayerController.uc
@@ -45,7 +45,7 @@ function PlayerTick( float DeltaTime )
 	super.PlayerTick( DeltaTime );
 	
 	if ( Pawn != none ) {
-		DrCamera( PlayerCamera ).CurrentCamera.SetTargetHeight( HeightHint );
+		DrCamera( PlayerCamera ).CurrentCamera.SetTargetHeight( HeightHint + Pawn.Location.Z );
 	}
 }
 
@@ -77,15 +77,15 @@ function UpdateRotation( float DeltaTime )
     local Vector MouseVec;
 
     MouseInput = DrMouseInput( PlayerInput );
-    MouseVec.X =  MouseInput.MousePos.X - myHUD.SizeX / 2 ;
-    MouseVec.Y =  MouseInput.MousePos.Y - myHUD.SizeY / 2;
+    MouseVec.X =  myHUD.SizeX / 2 - MouseInput.MousePos.X;
+    MouseVec.Y =  myHUD.SizeY / 2 - MouseInput.MousePos.Y;
     MouseVec.Z = 0.0;
 
     ViewRotation = Rotator( MouseVec );
     ViewRotation.Yaw += RotationOffset;
 
     if ( Pawn != none ) {
-        Pawn.SetDesiredRotation(ViewRotation);
+        Pawn.SetDesiredRotation( ViewRotation );
     }
 
     DeltaRot = rot( 0, 0, 0 );
diff --git a/Development/Src/Tendrils/Classes/DrPointLight.uc b/Development/Src/Tendrils/Classes/DrPointLight.uc
new file mode 100644
index 0000000..ecc6089
--- /dev/null
+++ b/Development/Src/Tendrils/Classes/DrPointLight.uc
@@ -0,0 +1,67 @@
+class DrPointLight extends PointLightMovable;
+
+var float SphereRadius;
+var PointLightComponent PLC;
+
+event simulated PostBeginPlay()
+{
+	foreach ComponentList( class'PointLightComponent', PLC ) {
+		break;
+	}
+    //CollisionComponent.SetActorCollision( true, false, true );
+	if ( PLC != none ) {
+		CollisionComponent.SetScale( PLC.Radius / SphereRadius );
+	}
+}
+
+event Touch( Actor Other, PrimitiveComponent OtherComp, Vector HitLocation, Vector HitNormal )
+{
+	`log( "LIGHT TOUCH" @ Other );
+	if( DrRookiePawn( Other ) != None ) {
+		`log("touched player");
+	}
+}
+
+event UnTouch(Actor Other)
+{
+	//local NPlayer_Pawn P;
+	
+	//if( isInRange(Other) )
+	//	return;
+	//P = NPlayer_Pawn(Other);
+	//if( P != None)
+	//{
+	//	`log("removeing"@self);
+	//	P.removePointLight(self);	
+	//}
+	//`log("POINTlight"@self@"just UNtouched"@Other);
+}
+
+DefaultProperties
+{
+	bCollideActors=true
+    bCollideWorld=true
+    bBlockActors=false
+    bNoEncroachCheck=false
+	CollisionType=COLLIDE_TouchAll
+	SphereRadius=160.0
+
+	Begin Object Name=PointLightComponent0
+	    LightAffectsClassification=LAC_DYNAMIC_AND_STATIC_AFFECTING
+	    CastShadows=TRUE
+	    CastStaticShadows=false
+	    CastDynamicShadows=TRUE
+	    bForceDynamicLight=true
+	    UseDirectLightMap=FALSE
+	    LightingChannels=(BSP=false,Static=false,Dynamic=TRUE,bInitialized=TRUE)
+	End Object
+
+	Begin Object class='StaticMeshComponent' name=SphereCollision
+		CollideActors=true
+		BlockZeroExtent=true
+		BlockNonZeroExtent=true
+		StaticMesh=StaticMesh'EngineMeshes.Sphere'
+	End Object
+	CollisionComponent=SphereCollision
+	Components.Add(SphereCollision)
+}
diff --git a/Development/Src/Tendrils/Classes/DrRookiePawn.uc b/Development/Src/Tendrils/Classes/DrRookiePawn.uc
index 4c09d1f..6811ab8 100644
--- a/Development/Src/Tendrils/Classes/DrRookiePawn.uc
+++ b/Development/Src/Tendrils/Classes/DrRookiePawn.uc
@@ -8,6 +8,7 @@ var Actor CurrentRoom;
 //    //return false;
 //}
 
+
 event PostBeginPlay()
 {
 }
diff --git a/Development/Src/Tendrils/Classes/DrSectionRoom.uc b/Development/Src/Tendrils/Classes/DrSectionRoom.uc
index 3d14f83..43013a1 100644
--- a/Development/Src/Tendrils/Classes/DrSectionRoom.uc
+++ b/Development/Src/Tendrils/Classes/DrSectionRoom.uc
@@ -54,7 +54,5 @@ DefaultProperties
     bCollideActors=true
 	bBlockActors=true
 	BlockRigidBody=true
-    BlockZeroExtent=true
-    BlockNonZeroExtent=true
     CollisionType=COLLIDE_BlockAll
 }
diff --git a/Development/Src/obj/Win32/Debug/Tendrils.ucproj.UnrealScriptGeneratedFileListAbsolute.txt b/Development/Src/obj/Win32/Debug/Tendrils.ucproj.UnrealScriptGeneratedFileListAbsolute.txt
index 2be685e..31ab3be 100644
--- a/Development/Src/obj/Win32/Debug/Tendrils.ucproj.UnrealScriptGeneratedFileListAbsolute.txt
+++ b/Development/Src/obj/Win32/Debug/Tendrils.ucproj.UnrealScriptGeneratedFileListAbsolute.txt
@@ -1,19 +1,19 @@
-I:\UDK-2013\UDKGame\Script\Core.u
-I:\UDK-2013\UDKGame\Script\Engine.u
-I:\UDK-2013\UDKGame\Script\IpDrv.u
-I:\UDK-2013\UDKGame\Script\GFxUI.u
-I:\UDK-2013\UDKGame\Script\GameFramework.u
-I:\UDK-2013\UDKGame\Script\UnrealEd.u
-I:\UDK-2013\UDKGame\Script\GFxUIEditor.u
-I:\UDK-2013\UDKGame\Script\WinDrv.u
-I:\UDK-2013\UDKGame\Script\OnlineSubsystemPC.u
-I:\UDK-2013\UDKGame\Script\OnlineSubsystemSteamworks.u
-I:\UDK-2013\UDKGame\Script\OnlineSubsystemGameCenter.u
-I:\UDK-2013\UDKGame\Script\SubstanceAir.u
-I:\UDK-2013\UDKGame\Script\SubstanceAirEd.u
-I:\UDK-2013\UDKGame\Script\UDKBase.u
-I:\UDK-2013\UDKGame\Script\UTEditor.u
-I:\UDK-2013\UDKGame\Script\UDN.u
-I:\UDK-2013\UDKGame\Script\UTGame.u
-I:\UDK-2013\UDKGame\Script\UTGameContent.u
-I:\UDK-2013\UDKGame\Script\Tendrils.u
+C:\UDK\UDK-2013-07\UDKGame\Script\Core.u
+C:\UDK\UDK-2013-07\UDKGame\Script\Engine.u
+C:\UDK\UDK-2013-07\UDKGame\Script\IpDrv.u
+C:\UDK\UDK-2013-07\UDKGame\Script\GFxUI.u
+C:\UDK\UDK-2013-07\UDKGame\Script\GameFramework.u
+C:\UDK\UDK-2013-07\UDKGame\Script\UnrealEd.u
+C:\UDK\UDK-2013-07\UDKGame\Script\GFxUIEditor.u
+C:\UDK\UDK-2013-07\UDKGame\Script\WinDrv.u
+C:\UDK\UDK-2013-07\UDKGame\Script\OnlineSubsystemPC.u
+C:\UDK\UDK-2013-07\UDKGame\Script\OnlineSubsystemSteamworks.u
+C:\UDK\UDK-2013-07\UDKGame\Script\OnlineSubsystemGameCenter.u
+C:\UDK\UDK-2013-07\UDKGame\Script\SubstanceAir.u
+C:\UDK\UDK-2013-07\UDKGame\Script\SubstanceAirEd.u
+C:\UDK\UDK-2013-07\UDKGame\Script\UDKBase.u
+C:\UDK\UDK-2013-07\UDKGame\Script\UTEditor.u
+C:\UDK\UDK-2013-07\UDKGame\Script\UDN.u
+C:\UDK\UDK-2013-07\UDKGame\Script\UTGame.u
+C:\UDK\UDK-2013-07\UDKGame\Script\UTGameContent.u
+C:\UDK\UDK-2013-07\UDKGame\Script\Tendrils.u
diff --git a/UDKGame/Content/Maps/tendrils/PlaceTest.udk b/UDKGame/Content/Maps/tendrils/PlaceTest.udk
index fe2226f..9e0b210 100644
Binary files a/UDKGame/Content/Maps/tendrils/PlaceTest.udk and b/UDKGame/Content/Maps/tendrils/PlaceTest.udk differ
diff --git a/UDKGame/Content/Maps/tendrils/Section03.udk b/UDKGame/Content/Maps/tendrils/Section03.udk
index 220b5c6..0e100c6 100644
Binary files a/UDKGame/Content/Maps/tendrils/Section03.udk and b/UDKGame/Content/Maps/tendrils/Section03.udk differ
